qt/preferences_dialog.cpp

```diff
@@ -78,22 +78,37 @@ namespace qt
       });
     }
 
+    QCheckBox * a3dAlwaysCheckBox = new QCheckBox("3D view always (experimental)");
+    QCheckBox * a3dInNavigationCheckBox = new QCheckBox("3D view during navigation");
     QCheckBox * a3dBuildingsCheckBox = new QCheckBox("3D buildings");
-    QCheckBox * a3dCheckBox = new QCheckBox("Perspective view (during navigation)");
     { 
-      bool allow3d, allow3dBuildings;
-      framework.Load3dMode(allow3d, allow3dBuildings);
+      {
+      bool allow3dAlways, allow3dInNavigation, allow3dBuildings;
+      framework.Load3dMode(allow3dAlways, allow3dInNavigation, allow3dBuildings);
+      a3dAlwaysCheckBox->setChecked(allow3dAlways);
+      a3dInNavigationCheckBox->setChecked(allow3dInNavigation);
       a3dBuildingsCheckBox->setChecked(allow3dBuildings);
-      a3dCheckBox->setChecked(allow3d);
-      connect(a3dBuildingsCheckBox, &QCheckBox::stateChanged, [&framework, &allow3d, &allow3dBuildings](int i)
+      }
+      connect(a3dAlwaysCheckBox, &QCheckBox::stateChanged, [&framework](int i)
       {
-        allow3dBuildings = static_cast<bool>(i);
-        framework.Allow3dMode(allow3d, allow3dBuildings);
+        bool allow3dAlways, allow3dInNavigation, allow3dBuildings;
+        framework.Load3dMode(allow3dAlways, allow3dInNavigation, allow3dBuildings);
+        allow3dAlways = static_cast<bool>(i);
+        framework.Allow3dMode(allow3dAlways, allow3dInNavigation, allow3dBuildings);
       });
-      connect(a3dCheckBox, &QCheckBox::stateChanged, [&framework, &allow3d, &allow3dBuildings](int i)
+      connect(a3dInNavigationCheckBox, &QCheckBox::stateChanged, [&framework](int i)
       {
-        allow3d = static_cast<bool>(i);
-        framework.Allow3dMode(allow3d, allow3dBuildings);
+        bool allow3dAlways, allow3dInNavigation, allow3dBuildings;
+        framework.Load3dMode(allow3dAlways, allow3dInNavigation, allow3dBuildings);
+        allow3dInNavigation = static_cast<bool>(i);
+        framework.Allow3dMode(allow3dAlways, allow3dInNavigation, allow3dBuildings);
+      });
+      connect(a3dBuildingsCheckBox, &QCheckBox::stateChanged, [&framework](int i)
+      {
+        bool allow3dAlways, allow3dInNavigation, allow3dBuildings;
+        framework.Load3dMode(allow3dAlways, allow3dInNavigation, allow3dBuildings);
```

Comment:
Would it make sense to decompose Load3dMode/Allow3dMode into Load3dAlways, Load3dBuildings, etc.?

Now it seems like there is a lot of redundant code just because these settings are coupled.

Reply:
It would make sense, and I agree this is a bit messy. But this PR is already pretty big and I would rather not start such a refactoring now. Perhaps in a later PR, provided that this even makes it to master.